src/main/resources/assets/js/components/Marathon.jsx

```diff
@@ -407,22 +459,33 @@ var Marathon = React.createClass({
           scaleApp={this.scaleApp}
           suspendApp={this.suspendApp}
           tasksFetchState={this.state.tasksFetchState}
+          router={this.props.router}
           ref="modal" />
       );
-    } else if (this.state.modalClass === NewAppModalComponent) {
-      modal = (
-        <NewAppModalComponent
-          model={this.state.activeApp}
-          onDestroy={this.handleModalDestroy}
-          onCreate={this.handleAppCreate}
-          ref="modal" />
-      );
-    } else if (this.state.modalClass === AboutModalComponent) {
-      modal = (
-        <AboutModalComponent
-          onDestroy={this.handleModalDestroy}
-          ref="modal" />
-      );
+    }
+  },
+
+  routeNewapp: function () {
+    return (
+      <NewAppModalComponent
+        model={this.state.collection.get("/" + this.state.activeAppId)}
```

Comment:
Why does this need a model? It creates a new one it self, as far as I can see?
https://github.com/mesosphere/marathon/blob/master/src/main/resources/assets/js/components/NewAppModalComponent.jsx#L33